### PR TITLE
feat(client): add `.createManyAndReturn()` top level method

### PR DESCRIPTION
**Pull request by Jolg42**
Based on https://github.com/prisma/prisma-engines/pull/4842

Contributes to https://github.com/prisma/team-orm/issues/1107

Related to https://github.com/prisma/prisma/issues/8131

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/not-so-exhaustive-schema/__snapshots__/binary.test.ts.snap [hunk 1]
```diff
@@ -13853,6 +14203,22 @@ export namespace Prisma {
       args?: SelectSubset<T, DCreateManyArgs<ExtArgs>>
     ): Prisma.PrismaPromise<BatchPayload>
 
+    /**
+     * Create many Ds and returns the data saved in the database.
+     *     @param {DCreateManyAndReturnArgs} args - Arguments to create many Ds.
```
Comment (SevInf) [inline packages/client/src/__tests__/integration/happy/not-so-exhaustive-schema/__snapshots__/binary.test.ts.snap:14208]: Should it be aligned this way?

**Review by SevInf — CHANGES_REQUESTED** [review]
Great job, only a few things need to be added/changed:

- Type tests for new code are now done in functional test setup rather than `tsd`
- I think we'll need a functional test that ensures `createManyAndReturn` is not present on unsupported providers, both on type and runtime levels.

File: packages/client/src/__tests__/integration/happy/not-so-exhaustive-schema/__snapshots__/binary.test.ts.snap (hunk 1 above)
Comment (Jolg42) [reply packages/client/src/__tests__/integration/happy/not-so-exhaustive-schema/__snapshots__/binary.test.ts.snap]: Fixed

File: packages/client/src/__tests__/types/createManyAndReturn/index.test-d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { expectError } from 'tsd'
```
[the remark this answers is not in the document]
Comment (Jolg42) [reply packages/client/src/__tests__/types/createManyAndReturn/index.test-d.ts]: I just deleted this instead.
We already have a test in the functional test suite as `packages/client/tests/functional/methods/createManyAndReturn/_matrix.ts`

**Review by SevInf — APPROVED** [review]

**Comment by Jolg42** [comment]
Note: This can be tried out in our dev version `5.14.0-dev.48` and up, it will be part of the official release on May 14th.

**Comment by Jolg42** [comment]
Thanks @marcus13371337 for trying it out!
I think what you mention here is that you would like support for creating nested relations for `.createMany()` and `.createManyAndReturn()`?
There is a tracking issue about that if you are interested: https://github.com/prisma/prisma/issues/5455

**Comment by marcus13371337** [comment]
@Jolg42 No not really, my point was that it would be nice if the return types (and some arguments) of `createManyAndReturn` and `findMany` gave you compatible results. I.e, `items` in this example would be of the same type!
```ts
const items = await prisma.myModel.createManyAndReturn({
  data: [...],
  select: { id: true, name: true }, 
  include: { children: true }
})

const items = await prisma.myModel.findMany({
  where: { ... },
  select: { id: true, name: true },
  include: { children: true }
})
```

**Comment by Jolg42** [comment]
@marcus13371337 You can now try `5.14.0-dev.61` and up, I fixed the types, `select` and `include` (and `omit` is the preview feature flag is set) now work 🎉

**Comment by Jolg42** [comment]
@marcus13371337 Thanks for this, we will check this, it might take a few days, there are some holidays planned in our team 📆 
Expect some news early next week.

Note: `include` `supplier` from the `voucher` side works.
```
      const vouchers = await prisma.voucher.createManyAndReturn({
        data: [
          {
            legalEntityId: legalEntity1.id,
            supplierId: suppliers[0].id,
          },
        ],
        include: {
          legalEntity: true,
          supplier: true,
        },
      })
```

**Pull request merged**